scripts/generate_example_previews.py

```
@@ -0,0 +1,136 @@
+                # "networkidle" is flaky by nature (Playwright's own docs warn
+                # against relying on it) and reliably reproduced spurious 404s
+                # here after a few pages' worth of prior navigations -- "load"
+                # is deterministic for these static, no-polling pages. The
+                # retry loop stays as a second line of defense.
+                for _attempt in range(3):
+                    response = page.goto(url, wait_until="load")
+                    if response and response.status == 200:
+                        break
+                if not response or response.status != 200:
+                    status = response.status if response else "no response"
+                    raise SystemExit(f"{slug}: giving up after 3 attempts ({status}) on {url}")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)